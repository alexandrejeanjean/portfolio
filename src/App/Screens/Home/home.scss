@import '../../constants/colors.scss';

.home-wrapper {
  .home-title-wrapper {
    display: flex;
    justify-content: center;
    flex-direction: column;
    position: relative;
    top: 14vh;
    left: 9vw;
    margin: auto;
    z-index: 10;

    .home-title-content {
      .home-title {
        margin-bottom: -0.5rem;
        font-size: 7em !important;
        color: $fourth;
      }

      .home-subtitle {
        width: 30vw;
        color: $tertiary;
        line-height: 30px;
      }
    }
  }

  .home-bubble-wrapper {
    .home-bubble,
    .home-bubble-arm,
    .home-bubble-round {
      position: absolute;
      z-index: 0;
    }

    .home-bubble {
      width: 70%;
      top: 0;
      left: 34%;
      right: 0;
      margin: auto;
      z-index: 1;
    }
  }

  @media screen and (max-width: 768px) {
    .home-title-wrapper {
      top: 32vh;
      left: 5vw;
      .home-title-content {
        .home-title {
          font-size: 3rem !important;
        }
        .home-subtitle {
          width: 80vw;
          font-size: 1.2rem;
          line-height: 28px;
        }
      }
    }

    .home-bubble-wrapper {
      .home-bubble,
      .home-bubble-arm,
      .home-bubble-round {
        position: absolute;
        z-index: 0;
      }

      .home-bubble {
        width: 120%;
        top: 5%;
        left: -10%;
      }
    }
  }
}

.head-animation {
  animation-name: headAnimation;
  animation-duration: 7s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
}

.under-head-animation {
  animation-name: underHeadAnimation;
  animation-duration: 7s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
}

.arm-animation {
  animation-name: armAnimation;
  animation-duration: 6s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
}

.left-arm {
  animation-delay: 0s;
  animation-name: leftArmAnimation;
  animation-duration: 6s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
}

.middle-arm {
  animation-delay: 0.4s;
}

.right-arm {
  animation-delay: 0.8s;
}

.bubble-animation {
  animation-name: bubbleAnimation;
  animation-duration: 6s;
  animation-fill-mode: both;
  animation-iteration-count: infinite;
}

@keyframes leftArmAnimation {
  0% {
    transform: translateY(-60px);
  }
  50% {
    transform: translateY(0px);
  }
  100% {
    transform: translateY(-60px);
  }
}

@keyframes armAnimation {
  0% {
    transform: translateY(-40px);
  }
  50% {
    transform: translateY(0px);
  }
  100% {
    transform: translateY(-40px);
  }
}

@keyframes headAnimation {
  0% {
    transform: translateY(-40px);
  }
  50% {
    transform: translateY(0px);
  }
  100% {
    transform: translateY(-40px);
  }
}

@keyframes underHeadAnimation {
  0% {
    transform: translateY(-30px);
  }
  50% {
    transform: translateY(0px);
  }
  100% {
    transform: translateY(-30px);
  }
}

@keyframes bubbleAnimation {
  0% {
    transform: translateY(40px);
  }
  50% {
    transform: translateY(0px);
  }
  100% {
    transform: translateY(40px);
  }
}
