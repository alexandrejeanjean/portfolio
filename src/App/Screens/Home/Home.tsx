import React from 'react'
// import BubbleT from '../../Assets/BubbleT.svg'
import ClassicButton from '../../Shared/Button/Button'
import './home.scss'

const Home = () => {
  return (
    <>
      <main className='home-wrapper page'>
        <section>
          <div className='home-title-wrapper'>
            <div className='home-title-content'>
              <h1 className='home-title'>
                Cødeur <br />
                du web.
              </h1>
              <p className='home-subtitle'>
                Bonjour, je m'appelle Alexandre. <br /> Depuis plus de 2 ans,
                j'agite de la matière grise pour donner vie à de nouvelles
                applications web & mobile innovantes.
              </p>
              <ClassicButton text='Suivez-moi' endpoint='/a-propos' />
            </div>
          </div>
          <div className='home-bubble-wrapper'>
            {/* <img src={BubbleT} className='home-bubble' alt='' /> */}
          </div>
        </section>
      </main>
    </>
  )
}

export default Home
