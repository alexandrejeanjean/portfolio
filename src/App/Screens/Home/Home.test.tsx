import React from 'react'
import { shallow, mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import Home from './Home'

let wrapper: any

beforeEach(() => {
  wrapper = shallow(<Home />)
})

describe('Home view', () => {
  it('should render home component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should be possible to click button to access next page', () => {
    const component = mount(
      <Router>
        <Home />
      </Router>
    )
    component.find('a.classic-btn').simulate('keydown', { keyCode: 32 })
    expect(component).toMatchSnapshot()
    component.unmount()
  })
})
