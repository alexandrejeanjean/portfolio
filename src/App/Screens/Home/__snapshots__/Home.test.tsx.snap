// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view should be possible to click button to access next page 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Home>
      <main
        className="home-wrapper page"
      >
        <section>
          <div
            className="home-title-wrapper"
          >
            <div
              className="home-title-content"
            >
              <h1
                className="home-title"
              >
                Cødeur 
                <br />
                du web.
              </h1>
              <p
                className="home-subtitle"
              >
                Bonjour, je m'appelle Alexandre. 
                <br />
                 Depuis plus de 2 ans, j'agite de la matière grise pour donner vie à de nouvelles applications web & mobile innovantes.
              </p>
              <ClassicButton
                endpoint="/a-propos"
                text="Suivez-moi"
              >
                <Link
                  className="classic-btn "
                  to="/a-propos"
                >
                  <LinkAnchor
                    className="classic-btn "
                    href="/a-propos"
                    navigate={[Function]}
                  >
                    <a
                      className="classic-btn "
                      href="/a-propos"
                      onClick={[Function]}
                    >
                      Suivez-moi
                    </a>
                  </LinkAnchor>
                </Link>
              </ClassicButton>
            </div>
          </div>
          <div
            className="home-bubble-wrapper"
          >
            <SvgComponent
              className="home-bubble"
            >
              <svg
                className="home-bubble"
                viewBox="0 0 1429 1109"
              >
                <linearGradient
                  gradientTransform="matrix(.9987 .05007 -.05007 .9987 33.22 -32.279)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__a"
                  x1={596.404}
                  x2={769.118}
                  y1={726.248}
                  y2={553.534}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <path
                  className="arm-animation middle-arm"
                  d="M631 486.3c13.3-18.5 49.6-23.7 68.5-8.6 7.6 6.1 11.8 15 14.6 28.8 7.8 38.6-5.9 65.9-15.1 101.4-10 38.7-9.8 68.8-9.6 111.7.4 70.7 14.9 109.4 6.4 112.5-9.5 3.4-34.8-41.9-48.4-72.5-13.5-30.3-28-63-26.8-105.5 1.2-42.2 16.4-42.7 16.3-85.3-.2-47.1-18.8-64.5-5.9-82.5z"
                  fill="url(#prefix__a)"
                  opacity={0.35}
                />
                <linearGradient
                  gradientTransform="matrix(.9987 .05007 -.05007 .9987 33.22 -32.279)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__b"
                  x1={588.404}
                  x2={761.118}
                  y1={721.248}
                  y2={548.534}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <path
                  className="arm-animation middle-arm"
                  d="M623.3 480.9c13.3-18.5 49.6-23.7 68.5-8.6 7.6 6.1 11.8 15 14.6 28.8 7.8 38.6-5.9 65.9-15.1 101.4-10 38.7-9.8 68.8-9.6 111.7.4 70.7 14.9 109.4 6.4 112.5-9.5 3.4-34.8-41.9-48.4-72.5-13.5-30.3-28-63-26.8-105.5 1.2-42.2 16.4-42.7 16.3-85.3-.2-47.1-18.9-64.4-5.9-82.5z"
                  fill="url(#prefix__b)"
                />
                <linearGradient
                  gradientTransform="matrix(.9994 .03419 -.03419 .9994 22.047 -25.575)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__c"
                  x1={689.962}
                  x2={862.676}
                  y1={698.393}
                  y2={525.679}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <path
                  className="arm-animation right-arm"
                  d="M722.5 458.8c13-18.7 49.3-24.5 68.4-9.7 7.7 6 12.1 14.8 15 28.5 8.4 38.5-4.9 66-13.5 101.6-9.4 38.9-8.8 68.9-7.8 111.8 1.5 70.7 16.7 109.1 8.2 112.3-9.5 3.6-35.5-41.4-49.6-71.7-14-30.1-29-62.5-28.4-105 .5-42.2 15.8-43 14.9-85.5-1-47-19.9-64.1-7.2-82.3z"
                  fill="url(#prefix__c)"
                  opacity={0.35}
                />
                <linearGradient
                  gradientTransform="matrix(.9994 .03419 -.03419 .9994 22.047 -25.575)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__d"
                  x1={681.962}
                  x2={854.676}
                  y1={693.393}
                  y2={520.679}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <path
                  className="arm-animation right-arm"
                  d="M714.6 453.5c13-18.7 49.3-24.5 68.4-9.7 7.7 6 12.1 14.8 15 28.5 8.4 38.5-4.9 66-13.5 101.6-9.4 38.9-8.8 68.9-7.8 111.8 1.5 70.7 16.7 109.1 8.2 112.3-9.5 3.6-35.5-41.4-49.6-71.7-14-30.1-29-62.5-28.4-105 .5-42.2 15.8-43 14.9-85.5-.9-47-19.8-64-7.2-82.3z"
                  fill="url(#prefix__d)"
                />
                <linearGradient
                  gradientUnits="userSpaceOnUse"
                  id="prefix__e"
                  x1={497.404}
                  x2={670.118}
                  y1={760.248}
                  y2={587.534}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <path
                  className="arm-animation left-arm"
                  d="M524 522c12.3-19.1 48.4-26.1 68-12 7.9 5.7 12.6 14.4 16 28 9.7 38.2-2.6 66.2-10 102-8.1 39.2-6.4 69.2-4 112 3.9 70.6 20.4 108.5 12 112-9.3 3.9-36.9-40.1-52-70-15-29.6-31.1-61.5-32-104-.9-42.2 14.3-43.5 12-86-2.5-47-22-63.3-10-82z"
                  fill="url(#prefix__e)"
                  opacity={0.35}
                />
                <linearGradient
                  gradientUnits="userSpaceOnUse"
                  id="prefix__f"
                  x1={489.404}
                  x2={662.118}
                  y1={755.248}
                  y2={582.534}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <path
                  className="arm-animation left-arm"
                  d="M516 517c12.3-19.1 48.4-26.1 68-12 7.9 5.7 12.6 14.4 16 28 9.7 38.2-2.6 66.2-10 102-8.1 39.2-6.4 69.2-4 112 3.9 70.6 20.4 108.5 12 112-9.3 3.9-36.9-40.1-52-70-15-29.6-31.1-61.5-32-104-.9-42.2 14.3-43.5 12-86-2.5-47-22-63.3-10-82z"
                  fill="url(#prefix__f)"
                />
                <linearGradient
                  gradientTransform="rotate(134.999 1024.401 436.094)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__g"
                  x1={1271.827}
                  x2={1315.88}
                  y1={1105.515}
                  y2={1061.461}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <circle
                  className="bubble-animation"
                  cx={376.1}
                  cy={168.9}
                  fill="url(#prefix__g)"
                  opacity={0.39}
                  r={31.2}
                />
                <linearGradient
                  gradientTransform="rotate(134.999 1024.401 436.094)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__h"
                  x1={1622.716}
                  x2={1688.796}
                  y1={955.99}
                  y2={889.909}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <circle
                  className="bubble-animation"
                  cx={233.7}
                  cy={538.3}
                  fill="url(#prefix__h)"
                  opacity={0.39}
                  r={46.7}
                />
                <linearGradient
                  gradientTransform="rotate(134.999 1024.401 436.094)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__i"
                  x1={1019.089}
                  x2={1041.116}
                  y1={308.487}
                  y2={286.46}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <circle
                  className="bubble-animation"
                  cx={1118.4}
                  cy={538.1}
                  fill="url(#prefix__i)"
                  opacity={0.39}
                  r={15.6}
                />
                <linearGradient
                  gradientTransform="rotate(134.999 1024.401 436.094)"
                  gradientUnits="userSpaceOnUse"
                  id="prefix__j"
                  x1={763.707}
                  x2={851.815}
                  y1={536.691}
                  y2={448.584}
                >
                  <stop
                    offset={0}
                    stopColor="#daf8ff"
                  />
                  <stop
                    offset={0.999}
                    stopColor="#b3d4fc"
                  />
                </linearGradient>
                <circle
                  className="bubble-animation"
                  cx={1137.6}
                  cy={242.9}
                  fill="url(#prefix__j)"
                  opacity={0.39}
                  r={62.3}
                />
                <g>
                  <linearGradient
                    gradientUnits="userSpaceOnUse"
                    id="prefix__k"
                    x1={328.11}
                    x2={1061.884}
                    y1={442.065}
                    y2={442.065}
                  >
                    <stop
                      offset={0}
                      stopColor="#daf8ff"
                    />
                    <stop
                      offset={0.999}
                      stopColor="#b3d4fc"
                    />
                  </linearGradient>
                  <path
                    className="head-animation"
                    d="M330.1 468.2c17.4-144.8 156-253.7 286.7-286.7 27.9-7 111.4-26.9 207.9 6.3C928.4 223.5 1049.3 321.3 1061 465c6.7 82.5-20.3 209.6-100.8 242.6-106.2 43.5-194.7-122.3-365.4-94.5-99.8 16.3-135.6 83.7-192.2 59.9-61.1-25.8-81.4-130.8-72.5-204.8z"
                    fill="url(#prefix__k)"
                  />
                  <linearGradient
                    gradientUnits="userSpaceOnUse"
                    id="prefix__l"
                    x1={342.11}
                    x2={1075.884}
                    y1={450.065}
                    y2={450.065}
                  >
                    <stop
                      offset={0}
                      stopColor="#daf8ff"
                    />
                    <stop
                      offset={0.999}
                      stopColor="#b3d4fc"
                    />
                  </linearGradient>
                  <path
                    className="under-head-animation"
                    d="M344.1 476.2c17.4-144.8 156-253.7 286.7-286.7 27.9-7 111.4-26.9 207.9 6.3C942.4 231.5 1063.3 329.3 1075 473c6.7 82.5-20.3 209.6-100.8 242.6-106.2 43.5-194.7-122.3-365.4-94.5-99.8 16.3-135.6 83.7-192.2 59.9-61.1-25.8-81.4-130.8-72.5-204.8z"
                    fill="url(#prefix__l)"
                    opacity={0.39}
                  />
                </g>
              </svg>
            </SvgComponent>
          </div>
        </section>
      </main>
    </Home>
  </Router>
</BrowserRouter>
`;

exports[`Home view should render home component correctly 1`] = `
<Fragment>
  <main
    className="home-wrapper page"
  >
    <section>
      <div
        className="home-title-wrapper"
      >
        <div
          className="home-title-content"
        >
          <h1
            className="home-title"
          >
            Cødeur 
            <br />
            du web.
          </h1>
          <p
            className="home-subtitle"
          >
            Bonjour, je m'appelle Alexandre. 
            <br />
             Depuis plus de 2 ans, j'agite de la matière grise pour donner vie à de nouvelles applications web & mobile innovantes.
          </p>
          <ClassicButton
            endpoint="/a-propos"
            text="Suivez-moi"
          />
        </div>
      </div>
      <div
        className="home-bubble-wrapper"
      >
        <SvgComponent
          className="home-bubble"
        />
      </div>
    </section>
  </main>
</Fragment>
`;
